src/dataDisplay/tooltip/tooltip.jsx

```diff
@@ -1,10 +1,243 @@
+     */
+    TransitionComponent: PropTypes.elementType,
+    /**
+     * Props applied to the transition element.
+     * By default, the element is based on this
+     * [`Transition`](http://reactcommunity.org/react-transition-group/transition) component.
+     */
+    TransitionProps: PropTypes.shape({}),
+};
+
+const defaultProps = {
+    arrow: false,
+    classes: undefined,
+    className: undefined,
+    disableFocusListener: false,
+    disableHoverListener: false,
+    disableTouchListener: false,
+    enterDelay: 100,
+    enterNextDelay: 0,
+    enterTouchDelay: 700,
+    id: undefined,
+    interactive: false,
+    leaveDelay: 0,
+    leaveTouchDelay: 1500,
+    onClose: undefined,
+    onOpen: undefined,
+    open: undefined,
+    placement: 'bottom',
+    PopperComponent: undefined,
+    PopperProps: {},
+    TransitionComponent: 'Grow',
+    TransitionProps: undefined,
+};
+
+const useStyles = makeStyles(({
+    palette,
+    shape,
+    spacing,
+    typography,
+}) => ({
+    tooltip: {
+        backgroundColor: palette.grey[500],
+        borderRadius: shape.borderRadius.main,
+        boxShadow: '0 2px 4px 0 rgba(0, 0, 0, .2)',
+        fontSize: typography.pxToRem(12),
+        margin: spacing(1),
+        padding: [[8, spacing(1)]],
+    },
+}));
+
+// eslint-disable-next-line prefer-arrow-callback
+const Tooltip = React.forwardRef(function Tooltip(props, ref) {
+    const {
+        arrow,
+        children,
+        classes: classesProp,
+        className,
+        disableFocusListener,
+        disableHoverListener,
+        disableTouchListener,
+        enterDelay,
+        enterNextDelay,
+        enterTouchDelay,
+        id,
+        interactive,
+        leaveDelay,
+        leaveTouchDelay,
+        onClose,
+        onOpen,
+        open,
+        placement,
+        PopperComponent,
+        title,
+        TransitionComponent,
+    } = props;
+
+    const classes = useStyles(props);
+
+    /**
+     * NOTE: These classNames are rendered on the child itself and not the tooltip.
+     */
+    const rootClasses = ClassNames(
+        BEM_TOOLTIP,
+        className,
+    );
+
+    return (
+        <MUITooltip
+            arrow={arrow}
+            classes={{
+                tooltip: classes.tooltip,
+                ...classesProp,
+            }}
+            className={rootClasses}
+            disableFocusListener={disableFocusListener}
+            disableHoverListener={disableHoverListener}
+            disableTouchListener={disableTouchListener}
+            enterDelay={enterDelay}
+            enterNextDelay={enterNextDelay}
+            enterTouchDelay={enterTouchDelay}
+            id={id}
+            interactive={interactive}
+            leaveDelay={leaveDelay}
+            leaveTouchDelay={leaveTouchDelay}
+            onClose={onClose}
+            onOpen={onOpen}
+            open={open}
+            placement={placement}
+            PopperComponent={PopperComponent}
```

Comment:
I think Material-UI used PascalCase for props for component type props. I really didn't look much into it other than copy and paste their stuff so that we could have some documentation.

Reply:
Got it.  I'm resolving all threads related to this.  All good; carry on.